packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -139,7 +153,7 @@ export interface NodegroupOptions {
    *
    * @default t3.medium
    */
-  readonly instanceType?: InstanceType;
+  readonly instanceType?: InstanceType[];
```

Comment:
Let's not break the api. Add another property called `additionalInstanceTypes` and mention that when `SPOT` is used, it is also recommended to provide additional types using this property.

Reply:
OK. So if we design the API like:

```ts
readonly instanceType: InstanceType;
readonly additonalInstanceTypes?: InstanceType[];
```

To create MNG with `t3.large`, `c5.large`, `m5.large` :

```ts
new Nodegroup(stack, id, {
   instanceType: new ec2.InstanceType('t3.large'),
   additionalInstanceTypes: [
      new ec2.InstanceType('c5.large'),
      new ec2.InstanceType('m5.large'),
  ],
  capacityType: eks.CapacityType.SPOT,
});
```

And we still determine the `AmiType` from `instanceType`, right?

Reply:
Well i'm not sure, what's going to happen if the AMI's are conflicting? I imagine this shouldn't happen often since the distinction is between graviton/gpu/inference instances.

What if we validate the all instance types resolve to the same AMI? and throw if they result in a conflict.

Reply:
Also i'm thinking we can probably go with `instanceTypes` (instead of `additionalInstanceTypes`), and deprecate `instanceType`. This would be clearer I think, and will map correctly to the underlying CFN properties.

Reply:
Yes. `instanceTypes` now.